src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -98,20 +101,40 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
                     LogMetricsFromAuthResult(authenticationResult, AuthenticationRequestParameters.RequestContext.Logger);
                     LogSuccessfulTelemetryToClient(authenticationResult, telemetryEventDetails, telemetryClients);
 
+#if NET6_0_OR_GREATER
+                    // Aggregates the successful requests based on client id, token source and cache refresh reason.
+                    OpenTelemetry.SuccessCounter.Add(1, 
+                        new (TelemetryConstants.ClientId, AuthenticationRequestParameters.AppConfig.ClientId), 
+                        new (TelemetryConstants.TokenSource, authenticationResult.AuthenticationResultMetadata.TokenSource), 
+                        new (TelemetryConstants.CacheInfoTelemetry, authenticationResult.AuthenticationResultMetadata.CacheRefreshReason));
+#endif
+
                     return authenticationResult;
                 }
                 catch (MsalException ex)
                 {
                     apiEvent.ApiErrorCode = ex.ErrorCode;
                     AuthenticationRequestParameters.RequestContext.Logger.ErrorPii(ex);
                     LogMsalErrorTelemetryToClient(ex, telemetryEventDetails, telemetryClients);
+#if NET6_0_OR_GREATER
```

Comment:
You can move this into the OpenTelemetry class or into some wrapper to avoid having `#if xyz` everywhere